### PR TITLE
Develop the Cost Settings Section

### PR DESCRIPTION
**Pull request by hieptl**
This PR fixes #

## Checklist
* [x] I have signed the CLA
* [x] I have updated/added any relevant documentation

### What's the goal of this PR?

* Develop the Cost Settings Section

### What changes did you make?

* Created the cost_settings.gohtml.
* Created the cost_settings.js
* Integrated with the API to load the list of cloud providers and instance types.

### What alternative solution should we consider, if any?

* None

### CODE REVIEW

**Comment by hieptl** [comment]
I need to update some logic for this PR, please do not review this PR now. Thank you!

**Comment by hieptl** [comment]
I updated the code. This PR is ready to be reviewed. Thank you!

**Comment by rbren** [comment]
A big issue I'm seeing is that we're not giving the user any feedback when there's an error. Some examples:
* email address is invalid
* API token is invalid
* terms of service box is unchecked

We need a way to surface issues here, even if it's just an `alert`

**Comment by rbren** [comment]
Can we run `trim` on the API token before saving it? When I copy/paste the one from the email, it has a space in front, which caused some confusion

**Comment by rbren** [comment]
A few more notes:
* We should change `Cost per CPU` to `Cost per CPU-hour` and `Cost per GB` to `Cost per GB-hour`.
* The placeholder text says `${X}/month`, which I think will lead people to type things like `$5/month`. Let's remove the placeholder text
* Can we use `<input type="number">` for the cost boxes? That way people don't type strings

**Comment by rbren** [comment]
Screen Shot 2022-08-30 at 10 07 51 AM
Screen Shot 2022-08-30 at 10 07 39 AM

**Comment by hieptl** [comment]
Hi @rbren : I updated the code based on your feedback. Thank you!

File: pkg/dashboard/templates/cost_settings.gohtml [hunk 1]
```diff
@@ -36,12 +36,12 @@
       </div>
       <div class="cost-settings-box__inputs-container">
         <label class="cost-settings-box__cost-per-cpu">
-          <span id="cost-settings-box__cost-per-cpu-label-content">Cost per CPU</span>
-          <input id="cost-settings-box__cost-per-cpu-input" placeholder="${X}/month" required disabled pattern="[0-9]+([\.,][0-9]+)?"/>
+          <span id="cost-settings-box__cost-per-cpu-label-content">Cost per CPU-hour</span>
+          <input id="cost-settings-box__cost-per-cpu-input" type="number" step="0.1" required disabled />
```
Comment (rbren) [inline pkg/dashboard/templates/cost_settings.gohtml:40]: Let's remove `step` - `0.1` is definitely too high, but I'm not sure if `.01` (one cent) is low enough).

File: pkg/dashboard/assets/js/api-token.js [hunk 2]
```diff
@@ -50,9 +50,19 @@
...
-      localStorage.setItem("apiKey", apiTokenInput.value);
+      const inputApiToken = apiTokenInput.value.trim();
+      fetch(
+        `${window.INSIGHTS_HOST}/v0/oss/instance-types?ossToken=${inputApiToken}`
+      ).then((response) => {
+        if (response && response.status !== 500) {
+          window.location.reload();
+          localStorage.setItem("apiKey", apiTokenInput.value.trim());
+        } else {
+          alert("Invalid API Token");
```
Comment (rbren) [inline pkg/dashboard/assets/js/api-token.js:63]: Do you want to do the new alert style in this PR, or a future one?

email

File: pkg/dashboard/templates/cost_settings.gohtml (hunk 1 above)
Comment (rbren) [reply pkg/dashboard/templates/cost_settings.gohtml:40]: Weirdly, this still shows up as `text` in firefox. No idea why

File: pkg/dashboard/assets/js/api-token.js (hunk 2 above)
Comment (hieptl) [reply pkg/dashboard/assets/js/api-token.js]: I updated the code. Thank you

**Review by rbren — APPROVED** [review]

**Pull request merged**